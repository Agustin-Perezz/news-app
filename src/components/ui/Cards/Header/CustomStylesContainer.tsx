import { styled } from '@mui/material/styles';
import MuiBox from '@mui/material/Box';

export const CustomBoxContainer = styled(MuiBox)(() => ({
  display: 'flex',
  justifyContent: 'flex-start',
  alignItems: 'center',
  margin: '12px auto',
  width: '280px',
  '@media (min-width: 576px)': {
    width: '585px',
  },
  '@media (min-width: 881px)': {
    width: '885px',
  },
}));
