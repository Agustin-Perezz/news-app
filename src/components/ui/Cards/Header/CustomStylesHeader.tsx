import { styled } from '@mui/material/styles';
import MuiBox from '@mui/material/Box';

export const CustomBoxHeader = styled(MuiBox)(() => ({
  display: 'flex',
  justifyContent: 'space-between',
  height: 'max-content',
  width: '95.5%',
  paddingLeft: '27px',
}));
