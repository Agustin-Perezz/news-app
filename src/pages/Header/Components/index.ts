export * from './CategoriesNavBar';
export * from './NavBar';
export * from './SearchBar';
