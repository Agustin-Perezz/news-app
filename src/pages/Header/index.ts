export * from './Components/NavBar';
export * from './Components/SearchBar';
export { default as HeaderPage } from './HeaderPage';
