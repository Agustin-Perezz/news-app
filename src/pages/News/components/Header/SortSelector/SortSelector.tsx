import { FormControl, InputLabel, MenuItem, Select } from '@mui/material';
import { useControllerSort } from '../../../hooks';
import { sortOptions } from './options-sort';

export const SortSelector: React.FC = () => {
  const { optionSort, onChange } = useControllerSort();

  return (
    <FormControl size="small" sx={{ m: 1, minWidth: 80 }}>
      <InputLabel>Sort by</InputLabel>
      <Select
        sx={{ padding: '5px 0px' }}
        label="Sort by"
        value={optionSort}
        onChange={onChange}
        autoWidth
      >
        {sortOptions.map((option) => (
          <MenuItem key={option} value={option}>
            {option}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};
