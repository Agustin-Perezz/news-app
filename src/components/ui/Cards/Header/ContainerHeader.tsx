import { Typography } from '@mui/material';
import { CustomBoxContainer } from './CustomStylesContainer';

type Props = { headerTitle?: string };

export const ContainerHeader: React.FC<Props> = ({ headerTitle }) => {
  return (
    <CustomBoxContainer>
      <Typography sx={{ fontSize: '18px' }}>{headerTitle}</Typography>
    </CustomBoxContainer>
  );
};
