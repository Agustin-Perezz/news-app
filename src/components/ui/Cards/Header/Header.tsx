import { Typography } from '@mui/material/';
import { CustomBoxHeader } from './CustomStylesHeader';
import { HeaderSort } from './Sort';

interface Props {
  cardsTitle: string;
}

export const Header: React.FC<Props> = ({ cardsTitle }) => {
  return (
    <CustomBoxHeader>
      <Typography variant="h6">{cardsTitle}</Typography>
      <HeaderSort />
    </CustomBoxHeader>
  );
};
