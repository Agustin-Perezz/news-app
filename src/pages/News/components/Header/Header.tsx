import { Typography } from '@mui/material/';
import { CustomBoxHeader } from './CustomStylesHeader';
import { SortSelector } from './SortSelector';

interface Props {
  headerTitle: string;
}

export const Header: React.FC<Props> = ({ headerTitle }) => {
  return (
    <CustomBoxHeader>
      <Typography sx={{ fontSize: '18px' }}>{headerTitle}</Typography>
      <SortSelector />
    </CustomBoxHeader>
  );
};
