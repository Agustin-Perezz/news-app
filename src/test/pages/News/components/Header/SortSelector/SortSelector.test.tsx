import { screen } from '@testing-library/react';
import { SortSelector } from '../../../../../../pages/News/components/Header/SortSelector';
import { renderWithCache } from '../../../../../utils';

type Props = { cacheValue: Record<string, string>; route: string };

describe('Test in <SortSelector />', () => {
  function setup(props: Props) {
    return renderWithCache({
      ...props,
      children: <SortSelector />,
    });
  }

  it('should match snapshot', () => {
    setup({ cacheValue: {}, route: '' });
    expect(screen).toMatchSnapshot();
  });
  it('should show the sort title and the default sort option', async () => {
    setup({ cacheValue: {}, route: '' });
    screen.getAllByText('Sort by');
    screen.getByText('Most Recent');
  });
  it('should show value from cache', () => {
    setup({ cacheValue: { 'option-value': 'Last Month' }, route: '' });
    screen.getByText('Last Month');
  });
});
