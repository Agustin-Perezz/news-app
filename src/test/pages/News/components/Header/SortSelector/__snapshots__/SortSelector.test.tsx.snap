// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SortSelector /> should match snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-os7j8m-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Sort by
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1lvwyej-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="button"
        tabindex="0"
      >
        Most Recent
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="Most Recent"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Sort by
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
